### PR TITLE
Extend the precast FireElemental checker for StormElemental

### PR DESCRIPTION
**Pull request by HawkCorrigan**
title.

Depends on the fact that Fire and Storm Elementals count as the only pets so it should works as long as there are no trinkets, azerite traits or w/e that spawn "pets" in the sense of the "byPlayerPet_damage" check.

AFAIK and what I checked the only thing I can think of is Echo Of The Elementals Trait whit which the logic still works because the casts are interlinked.

TBF Storm Elemental should not be cast precombat anyway, because it is a .05% damage loss, so it shouldn't be relevant for anyone who plays a proper rotation ^^

Checked with **https://www.warcraftlogs.com/reports/gxWQvyP9dbDtkHpw/#fight=3&source=24**

### CODE REVIEW

**Comment by HawkCorrigan** [comment]
Gonna need to fix a thing with ghuun trinket first before merging

**Comment by HawkCorrigan** [comment]
fixed.

File: src/parser/shaman/elemental/modules/features/StormFireElemental.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  relevantData = this.selectedCombatant.hasTalent(TALENTS.STORM_ELEMENTAL_TALENT.id) ? this.data.StormElemental : this.data.FireElemental;
+
+
+  on_byPlayerPet_damage(event){
+      if(this.last_pet_summon_timeStamp!==null) {
+        return;
+      }
+      if(!this.relevant.damageSpells.includes(event.ability.guid)) {
+        return;
```
Comment (Pewtro) [inline src/parser/shaman/elemental/modules/features/StormFireElemental.js]: Shouldn't this be called `!this.relevantData.damageSpells.includes(...)`

Comment (HawkCorrigan) [reply src/parser/shaman/elemental/modules/features/StormFireElemental.js]: done.
IntelliJ must have missed that for some reason

File: src/parser/shaman/elemental/modules/features/StormFireElemental.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  last_pet_summon_timeStamp=null;
+
+  relevantData = this.selectedCombatant.hasTalent(TALENTS.STORM_ELEMENTAL_TALENT.id) ? this.data.StormElemental : this.data.FireElemental;
+
+
+  on_byPlayerPet_damage(event){
+      if(this.last_pet_summon_timeStamp!==null) {
+        return;
+      }
```
Comment (Pewtro) [inline src/parser/shaman/elemental/modules/features/StormFireElemental.js]: lack of spaces here annoys me 😄

Comment (HawkCorrigan) [reply src/parser/shaman/elemental/modules/features/StormFireElemental.js]: do you mean free lines before and after the ifs?

Comment (Pewtro) [reply src/parser/shaman/elemental/modules/features/StormFireElemental.js]: In this ``      if(this.last_pet_summon_timeStamp!==null) {``

Comment (HawkCorrigan) [reply src/parser/shaman/elemental/modules/features/StormFireElemental.js]: ah aight ^^
I change that at a later point in time :P

**Pull request merged**